File: problem25/p25.py
def get_fibonacci_number(index):
    f1 = f2 = 1
    for i in range(3, index + 1, 2):
        f1 += f2
        f2 += f1
    if index % 2 == 0:
        return f2
    return f1


def a_thousand_digit_fibonacci_number(contained_digit):
    i = 1
    compare = 10 ** (contained_digit - 1)
    while True:
        if get_fibonacci_number(i) > compare:
            return i
        i += 1


print(a_thousand_digit_fibonacci_number(1000))

File: problem25/p25.java
import java.math.BigInteger;

public class p25 {
	
	public static BigInteger getFibonacciNumber(int index) {
		int i;
		BigInteger f1 = new BigInteger("1");
		BigInteger f2 = new BigInteger("2");
		
		for(i=3; i<index+1; i+=2) {
			f1 = f1.add(f2);
			f2 = f2.add(f1);
		}
		if(index % 2 == 0) {
			return f2;
		}
		return f1;
	}
	
	public static int a_thousand_digit_fibonacci_number(int contained_digit) {
		
		int i = 1;
		while(true) {
			if(getFibonacciNumber(i).toString().length() >= contained_digit) {
				return i+1;
			}
			i += 1;
		}
	}

	public static void main(String args[]) {
			
			System.out.println(a_thousand_digit_fibonacci_number(1000));
	
	}
	
}
